ironfist/src/cpp/adventure/adv.cpp

```diff
@@ -2,6 +2,57 @@
 #include "adventure/map.h"
 #include "game/game.h"
 #include "scripting/hook.h"
+#include "prefs.h"
+
+extern void __fastcall NormalDialog(char *, int, int, int, int, int, int, int, int, int);
+
+int advManager::ProcessDeSelect(struct tag_message *GUIMessage_evt, int *a3, class mapCell * *a4) {
+	if(GUIMessage_evt->yCoordOrFieldID == 4)
+		{
+		DWORD hero_reminder_reg_dword = read_pref<DWORD>("Show Hero Movement Reminder");
+		//default is true, but read_pref() returns -1 if the value is not set
+		bool show_hero_movement_reminder = !(hero_reminder_reg_dword == 0);
+		
+		if(!show_hero_movement_reminder || !gpCurPlayer->HasMobileHero() ||
+				(
+				NormalDialog("One or more heroes may still move, are you sure you want to end your turn?",
+							2, -1, -1, -1, 0, -1, 0, -1, 0),
+				gpWindowManager->buttonPressedCode != 30726)

```

Comment:
Yeah, h2guiview is not the best.

You should see another window where it's displaying the actual GUI. Seeing whatever is drawn at that y coordinate, width, and height tells you what it is.

There's a hash function in resources.cpp that converts filenames into an ID. It's saying that, whatever ICN file has that ID, it's the 7th and 8th images in that ICN.

Or, you can just

Also, if you dump the .rsrc section of HEROES2W.exe using PE Explorer, it will tell you all the GUID codes for things like e.g.: the File menu. I believe they like to share codes between that and the actual game GUI.